table/iterator.go

```diff
@@ -102,7 +105,11 @@ func (itr *blockIterator) Error() error {
 	return itr.err
 }
 
-func (itr *blockIterator) Close() {}
+func (itr *blockIterator) Close() {
+	if itr.compressed {
```

Comment:
this feels a bit weird that it's done in this way. Can't we keep a single buffer pool for both decompressed and regular data?